Monika After Story/game/script-topics.rpy

```diff
@@ -1819,41 +1819,339 @@ init 5 python:
     )
 
 label monika_rain_holdme:
-    # we only want this if it rains
-    if mas_is_raining or mas_isMoniAff(higher=True):
-        stop music fadeout 1.0
 
-        # clear selected track
-        $ songs.current_track = songs.FP_NO_SONG
-        $ songs.selected_track = songs.FP_NO_SONG
+    if mas_isMoniHappy(higher=True):
+        # happy or above
 
-        # hide ui and disable hotkeys
-        $ HKBHideButtons()
-        $ store.songs.enabled = False
+        # we only want this if it rains
+        if mas_is_raining or mas_isMoniAff(higher=True):
+            call monika_holdme_prep
+            m 1a "Of course, [player]."
+            call monika_holdme_start
 
-        m 1a "Of course, [player]."
-        show monika 6dubsa
-        $ ui.add(PauseDisplayable())
-        $ ui.interact()
+            call monika_holdme_reactions
+            # small affection increase so people don't farm affection with this one.
+            $ mas_gainAffection(modifier=0.25)
 
-        # renable ui and hotkeys
-        $ store.songs.enabled = True
-        $ HKBShowButtons()
-        # small affection increase so people don't farm affection with this one.
-        $ mas_gainAffection(modifier=0.25)
+            if mas_isMoniAff(higher=True):
+                m 1hubfb "You can hold me anytime you want, [player]."
+            else:
+                m 1hubfb "You can hold me anytime it rains, [player]."
 
-        if mas_isMoniAff(higher=True):
-            m 1hubfb "You can hold me anytime you want, [player]."
         else:
-            m 1hubfb "You can hold me anytime it rains, [player]."
+            # no affection loss here, doesn't make sense to have it
+            m 1rksdlc "..."
+            m 1rksdlc "The mood doesn't feel right, [player]."
+            m 1dsc "Sorry..."
+
+    elif mas_isMoniNormal():
```

Comment:
this topic can only be selected if affection is happy+. There's no point in having additional flows, which is why it was changed recently.